b2/sync/filters.py

```diff
@@ -0,0 +1,129 @@
+######################################################################
+#
+# File: b2/sync/filters.py
+#
+# Copyright 2016 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+import logging
+import re
+import six
+import os.path
+from abc import ABCMeta, abstractmethod
+
+logger = logging.getLogger(__name__)
+
+
+@six.add_metaclass(ABCMeta)
+class FileFilter(object):
+    @abstractmethod
+    def is_exclude_filter(self):
```

Comment:
An interface that lets you ask what type of object you have usually means that the abstraction is could be better.

A class docstring saying what the class does might help.  From a quick look at the code, I think that: (1) there are filters and a filter manager, (2) the filter manager is what combines filters, and (3) the filter manager is what client code will use when testing whether a file or directory should be included.  Is that right?